gtubt/lib/ui/components/hamburgerMenu.dart

```diff
@@ -43,35 +43,87 @@ class _HamburgerMenuComponentsState extends State<HamburgerMenuComponents> {
     );
   }
 
+  void _settingsButtonFunction() {
+    BlocProvider.of<PageBloc>(context).add(
+      PageChanged(
+        context: context,
+        routeName: SETTINGS_URL,
+      ),
+    );
+  }
+
+  Widget _buildProfileHeader(String username, String profileImage) {
+    return Column(
+      children: [
+        Container(
+            width: 150,
+            height: 150,
+            decoration: BoxDecoration(shape: BoxShape.circle),
+            clipBehavior: Clip.antiAlias,
+            child: FittedBox(
+              child: profileImage != null
+                  ? Image.network(
+                      profileImage,
+                    )
+                  : Icon(
+                      Icons.account_circle,
+                      color: ColorSets.pageBackgroundColor,
+                    ),
+            )),
+        Padding(
+          padding: const EdgeInsets.only(top: 12.0, bottom: 30.0),
+          child: Text(
+            username ?? '',
+            style:
+                TextStyles.subtitle1.copyWith(color: ColorSets.lightTextColor),
+          ),
+        )
+      ],
+    );
+  }
+
   Widget _buildHamburgerMenuItem(String menuItemName, dynamic function) {
     return InkWell(
       child: Container(
         margin: EdgeInsets.symmetric(vertical: 10),
         height: 50,
         color: ColorSets.profilePageThemeColor,
         child: Center(
-          child: Text(menuItemName, style: TextStyles.subtitle1.copyWith(color: ColorSets.lightTextColor),),
+          child: Text(
+            menuItemName,
+            style:
+                TextStyles.subtitle1.copyWith(color: ColorSets.lightTextColor),
+          ),
         ),
       ),
       onTap: () => function(),
     );
   }
 
+  Widget _buildSpacer(double spaceFactor){
+    var _height = MediaQuery.of(context).size.height;
+    return SizedBox(
+      height: _height*spaceFactor,
+    );
+  }
+
   Widget _buildHamburgerMenu() {
     List<Widget> _menuItems = List<Widget>();
 
-    if (_authBloc.state is AuthenticationAuthenticated) {
-      _menuItems.add(_buildHamburgerMenuItem('PROFILE FOTO', null));
-      _menuItems.add(_buildHamburgerMenuItem('USERNAME', null));
+    if (_authBloc.state is !AuthenticationAuthenticated) {
+      _menuItems.add(
+          _buildProfileHeader('Your Name', 'http://placekitten.com/400/400'));
```

Comment:
Bunlari test icin mi biraktin bilmiyorum ama `UserService.currentUser` dan alman gerekli burdaki bilgileri.

Reply:
`UserService.currentUser` authentication olmadigi durumda null olacak. 

is isterlerinde yanlislik olmus. burada logged in olmadigi icin hic bir sey gorunmemeli sadece giris yap ve ayarlar gorunmeli. logout butonu gorunuyor bu sekilde olmamali.

@hybrayhem is isterlerini guncelliyorum.

Reply:
> Bunlari test icin mi biraktin bilmiyorum ama `UserService.currentUser` dan alman gerekli burdaki bilgileri.

Evet, o kısımları değiştireceğim